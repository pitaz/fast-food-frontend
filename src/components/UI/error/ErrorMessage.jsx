import React from 'react';
import PropTypes from 'prop-types';

const ErrorMessage = ({errors}) => (
  (
    <span>
     <p className="error">{errors}</p>
    </span>
  )
);

ErrorMessage.propTypes = {
  errors: PropTypes.instanceOf(Array),
};
export default ErrorMessage;
