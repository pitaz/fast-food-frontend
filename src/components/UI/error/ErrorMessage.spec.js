/* eslint-disable react/no-children-prop */
/* eslint-disable no-undef */
import React from 'react';
import expect from 'expect';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import ErrorMessage  from './ErrorMessage.jsx';
import configureMockStore from 'redux-mock-store';


const mockStore = configureMockStore();
let wrapper;

describe('The Signin Component Test Suite', () => {
  wrapper = shallow(<ErrorMessage />);

  it('renders the ErrorMessage component', () => {
    expect(wrapper.exists()).toBe(true);
    expect(wrapper).toMatchSnapshot();
  }); 

  it('should match the snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render <ErrorMessageForm /> Component', () => {
    expect(wrapper.length).toBe(1);
    expect(wrapper.find(ErrorMessage)).toBeDefined();
  });
});
