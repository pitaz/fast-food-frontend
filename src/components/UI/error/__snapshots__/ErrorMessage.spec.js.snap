// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Signin Component Test Suite renders the ErrorMessage component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ErrorMessage />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <p
        className="error"
      />,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": undefined,
        "className": "error",
      },
      "ref": null,
      "rendered": null,
      "type": "p",
    },
    "type": "span",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <p
          className="error"
        />,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "className": "error",
        },
        "ref": null,
        "rendered": null,
        "type": "p",
      },
      "type": "span",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`The Signin Component Test Suite should match the snapshot 1`] = `
<span>
  <p
    className="error"
  />
</span>
`;
